lib/graphql/schema/printer.rb

```diff
@@ -60,7 +60,9 @@ def initialize(schema, context: nil, only: nil, except: nil, introspection: fals
 
       # Return the GraphQL schema string for the introspection type system
       def self.print_introspection_schema
-        query_root = ObjectType.define(name: "Root")
+        query_root = ObjectType.define(name: "Root") do
+          field :throwaway_field, types.String
```

Comment:
Adding this was needed in order to satisfy the new validation, but note that the Root type itself gets excluded from the `introspection_schema_ast` before it is stringified: https://github.com/chrisbutcher/graphql-ruby/blob/error-when-query-defined-but-empty/lib/graphql/schema/printer.rb#L70